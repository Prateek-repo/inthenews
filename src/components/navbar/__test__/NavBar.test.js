import { shallow, configure } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import toJson from "enzyme-to-json";

import Container from "../NavBar";

configure({ adapter: new Adapter() });

describe("<NavBar/>", () => {
  let wrapper;

  const defaultProps = {
    setTheCountry: jest.fn(),
    darkMode: jest.fn(),
  };

  beforeEach(() => {
    wrapper = shallow(<Container {...defaultProps} />);
  });

  it("SnapShot creation", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("handling handleChange", () => {
    const componentInstance = wrapper.instance();
    componentInstance.handleChange("us");
    expect(componentInstance.state.selectedOption).toEqual("us");
  });

  it("handling darkModeEnable", () => {
    wrapper.setState({
      darkModeStatus: true,
    });
    const componentInstance = wrapper.instance();
    componentInstance.darkModeEnable();
    expect(componentInstance.state.darkModeStatus).toEqual(false);
  });

  it("handling handleChange prop call", () => {
    const componentInstance = wrapper.instance();
    componentInstance.handleChange("us");
    expect(defaultProps.setTheCountry).toHaveBeenCalledTimes(2);
  });

  it("handling darkModeEnable prop call", () => {
    const componentInstance = wrapper.instance();
    componentInstance.darkModeEnable();
    expect(defaultProps.darkMode).toHaveBeenCalledTimes(2);
  });
});
