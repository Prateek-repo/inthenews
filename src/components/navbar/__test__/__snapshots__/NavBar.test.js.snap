// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar/> SnapShot creation 1`] = `
<div>
  <nav
    className="navbar fixed-top navbar-expand-lg navbar-light bg-light"
  >
    <div
      className="container-fluid"
    >
      <Link
        className="navbar-brand nav-link"
        to="/general"
      >
        <img
          alt=""
          height="24"
          src=""
          width="30"
        />
        Â 
        <b>
          InTheNews
        </b>
      </Link>
      <button
        aria-controls="navbarSupportedContent"
        aria-expanded="false"
        aria-label="Toggle navigation"
        className="navbar-toggler"
        data-bs-target="#navbarSupportedContent"
        data-bs-toggle="collapse"
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        id="navbarSupportedContent"
      >
        <ul
          className="navbar-nav me-auto mb-2 mb-lg-0"
        >
          <li
            className="nav-item"
          >
            <Link
              className="nav-link"
              to="/general"
            >
              General
            </Link>
          </li>
          <li
            className="nav-item"
          >
            <Link
              className="nav-link"
              to="/health"
            >
              Health
            </Link>
          </li>
          <li
            className="nav-item"
          >
            <Link
              className="nav-link"
              to="/business"
            >
              Business
            </Link>
          </li>
          <li
            className="nav-item"
          >
            <Link
              className="nav-link"
              to="/entertainment"
            >
              Entertainment
            </Link>
          </li>
          <li
            className="nav-item"
          >
            <Link
              className="nav-link"
              to="/sports"
            >
              Sports
            </Link>
          </li>
          <li
            className="nav-item"
          >
            <Link
              className="nav-link"
              to="/technology"
            >
              Technology
            </Link>
          </li>
          <li
            className="nav-item"
          >
            <Link
              className="nav-link"
              to="/science"
            >
              Science
            </Link>
          </li>
        </ul>
        <ul
          className="navbar-nav me-auto mb-2 mb-lg-0"
        >
          <li
            className="nav-item"
          >
            <ForwardRef
              onChange={[Function]}
              options={
                Array [
                  Object {
                    "label": "India",
                    "value": "in",
                  },
                  Object {
                    "label": "USA",
                    "value": "us",
                  },
                  Object {
                    "label": "United Kingdom",
                    "value": "gb",
                  },
                  Object {
                    "label": "United Arab Emirates",
                    "value": "ae",
                  },
                  Object {
                    "label": "Argentina",
                    "value": "ar",
                  },
                  Object {
                    "label": "Austria",
                    "value": "at",
                  },
                  Object {
                    "label": "Australia",
                    "value": "au",
                  },
                  Object {
                    "label": "Belgium",
                    "value": "be",
                  },
                  Object {
                    "label": "Bulgaria",
                    "value": "bg",
                  },
                  Object {
                    "label": "Brazil",
                    "value": "br",
                  },
                  Object {
                    "label": "Canada",
                    "value": "ca",
                  },
                  Object {
                    "label": "Switzerland",
                    "value": "ch",
                  },
                  Object {
                    "label": "China",
                    "value": "cn",
                  },
                  Object {
                    "label": "Colombia",
                    "value": "co",
                  },
                  Object {
                    "label": "Cuba",
                    "value": "cu",
                  },
                  Object {
                    "label": "Czech Republic",
                    "value": "cz",
                  },
                  Object {
                    "label": "Germany",
                    "value": "de",
                  },
                  Object {
                    "label": "Egypt",
                    "value": "eg",
                  },
                  Object {
                    "label": "France",
                    "value": "fr",
                  },
                  Object {
                    "label": "Greece",
                    "value": "gr",
                  },
                  Object {
                    "label": "Hong Kong",
                    "value": "hk",
                  },
                  Object {
                    "label": "Hungary",
                    "value": "hu",
                  },
                  Object {
                    "label": "Indonesia",
                    "value": "id",
                  },
                  Object {
                    "label": "Ireland",
                    "value": "ie",
                  },
                  Object {
                    "label": "Israel",
                    "value": "il",
                  },
                  Object {
                    "label": "Japan",
                    "value": "jp",
                  },
                  Object {
                    "label": "South Korea",
                    "value": "kr",
                  },
                  Object {
                    "label": "Lithuania",
                    "value": "lt",
                  },
                  Object {
                    "label": "Latvia",
                    "value": "lv",
                  },
                  Object {
                    "label": "Morocco",
                    "value": "ma",
                  },
                  Object {
                    "label": "Mexico",
                    "value": "mx",
                  },
                  Object {
                    "label": "Nigeria",
                    "value": "ng",
                  },
                  Object {
                    "label": "Netherlands",
                    "value": "nl",
                  },
                  Object {
                    "label": "Norway",
                    "value": "no",
                  },
                  Object {
                    "label": "New Zealand",
                    "value": "nz",
                  },
                  Object {
                    "label": "Philippines",
                    "value": "ph",
                  },
                  Object {
                    "label": "Poland",
                    "value": "pl",
                  },
                  Object {
                    "label": "Portugal",
                    "value": "pt",
                  },
                  Object {
                    "label": "Romania",
                    "value": "ro",
                  },
                  Object {
                    "label": "Serbia",
                    "value": "rs",
                  },
                  Object {
                    "label": "Russia",
                    "value": "ru",
                  },
                  Object {
                    "label": "Saudi Arabia",
                    "value": "sa",
                  },
                  Object {
                    "label": "Sweden",
                    "value": "se",
                  },
                  Object {
                    "label": "Singapore",
                    "value": "sg",
                  },
                  Object {
                    "label": "Slovenia",
                    "value": "si",
                  },
                  Object {
                    "label": "Slovakia",
                    "value": "sk",
                  },
                  Object {
                    "label": "Thailand",
                    "value": "th",
                  },
                  Object {
                    "label": "Turkey",
                    "value": "tr",
                  },
                  Object {
                    "label": "Taiwan",
                    "value": "tw",
                  },
                  Object {
                    "label": "Ukraine",
                    "value": "ua",
                  },
                  Object {
                    "label": "Venezuela",
                    "value": "ve",
                  },
                  Object {
                    "label": "South Africa",
                    "value": "za",
                  },
                ]
              }
              value="World"
            />
          </li>
          <li
            className="nav-item"
            style={
              Object {
                "marginLeft": "32px",
              }
            }
          >
            <button
              className="btn-sm btn-dark"
              onClick={[Function]}
              type="button"
            >
              Go Dark Mode
            </button>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</div>
`;
